### PR TITLE
[SRP] Add "2021-08-01" API version

### PR DESCRIPTION
**Pull request by zfchen95**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json
Comment (blueww) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:2843]: Please update example for the 2 new properties of “ActiveDirectoryProperties”

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json
Comment (blueww) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:1882]: Please indicate which password will be regenerated in the API description?
Will only SshPassword be regenerated? Or SharedKey also be regenerated? (SshAuthorizedKeys is set by user, per my understanding it should not be regenearte. Please confirm)

**Comment by evelyn-ys** [comment]
May I know why the python track2 SDK from CI pipeline doesn't container the new api version? Where can I found the private Python SDK for CLI prototype?

**Review by evelyn-ys — COMMENTED** [review]
@zfchen95 Can you update `readme.md` and add `Tag: package-2021-08` so that Python track2 SDK can be generated correctly?

**Comment by blueww** [comment]
@zfchen95 
Would you please add enable/disable SFTP feature?

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json
Comment (zfchen95) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:2852]: added in StorageAccountEnableAD.json

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json
Comment (zfchen95) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:1891]: It will only regenerate sshPassword. To regenerate sharedKey, user will need to set hasSharedKey to false and true.

**Comment by zfchen95** [comment]
> @zfchen95 Can you update `readme.md` and add `Tag: package-2021-08` so that Python track2 SDK can be generated correctly?

Done

**Comment by zfchen95** [comment]
> @zfchen95 Would you please add enable/disable SFTP feature?

Done

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json
Comment (blueww) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:1644]: Generate .net SDK failed with following error, it looks this Uri missing an end "}".

> ERROR: Schema violation: Additional properties not allowed: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccounts/{accountName}/localUsers
>     - https://raw.githubusercontent.com/zfchen95/azure-rest-api-specs/aug21/specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:1644:4 ($.paths["/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccounts/{accountName}/revokeUserDelegationKeys"])

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json
Comment (blueww) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:2951]: What's the default value if user not set it?

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json
Comment (blueww) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:4874]: Could you specify which properties of the permission scope is optional? Or they are all required?

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json
Comment (zfchen95) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:2951]: default interpretation is false if not specified

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json
Comment (zfchen95) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:4874]: they are all required. Updated the definition.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json
Comment (zfchen95) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:1644]: should be fixed now

**Comment by msyyc** [comment]
Hi @zfchen95  python configuration is not right and I commit right configuration again, please don't override the commit anymore:
image

**Comment by zfchen95** [comment]
All the required checks passed. Please let me know if there is any concern.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json
Comment (blueww) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:1764]: From .net SDK/Pwoershell point of view, We should remove the "default" responds handling part, since:
1. Without  the default part, the error will be handled by default CloudException in .net SDK, and the error details will be in exception message. 
2. With the default part, the error will be handle by storage defined "#/definitions/ErrorResponse", the returned exception in .net SDK will only has error code in exception message, but won't has detail error message in the exception message. It will be difficult for customer to find the detail error message, as they need go into the exception properties and  try to find which property contains the error.

As the default error handling can handle error with following body very well. I would suggest:

-  if the error body of an API have following format, we should not add "default" responds part for it.
- only add default responds handling for error , when the error body is not in following format.

```
Body:
{
  "error": {
    "code": "***",
    "message": "****"
  }
}
```

For you reference, following are errors from .net SDK:
Error from API without default responds section:
```
Microsoft.Rest.Azure.CloudException: An operation is currently performing on this storage account that requires exclusive access.
   at Microsoft.Azure.Management.Storage.StorageAccountsOperations.<DeleteWithHttpMessagesAsync>d__7.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Management.Storage.StorageAccountsOperationsExtensions.<DeleteAsync>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Management.Storage.StorageAccountsOperationsExtensions.Delete(IStorageAccountsOperations operations, String resourceGroupName, String accountName)
   at Microsoft.Azure.Commands.Management.Storage.RemoveAzureStorageAccountCommand.ExecuteCmdlet()
   at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord()
```
Error from API with default responds section:
```
Microsoft.Azure.Management.Storage.Models.ErrorResponseException: Operation returned an invalid status code 'BadRequest'
   at Microsoft.Azure.Management.Storage.StorageAccountsOperations.<BeginHierarchicalNamespaceMigrationWithHttpMessagesAsync>d__23.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.Azure.Management.Storage.StorageAccountsOperations.<HierarchicalNamespaceMigrationWithHttpMessagesAsync>d__17.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.Azure.Management.Storage.StorageAccountsOperationsExtensions.<HierarchicalNamespaceMigrationAsync>d__25.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at Microsoft.Azure.Management.Storage.StorageAccountsOperationsExtensions.HierarchicalNamespaceMigration(IStorageAccountsOperations operations, String resourceGroupName, String accountName, String requestType)
   at Microsoft.Azure.Commands.Management.Storage.InvokeAzureStorageAccountHierarchicalNamespaceUpgradeCommand.ExecuteCmdlet()
   at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord()
```

**Review by blueww — CHANGES_REQUESTED** [review]

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json
Comment (zfchen95) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:4928]: AFAIK, we only support Blob and File today. I did not model it as an enum in case we need to add another type in future.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json
Comment (zfchen95) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:1764]: Hi Wei, I added it back because it failed the validation if no default response is defined. 

"paths":[{"tag":"Location","path":"https://github.com/Azure/azure-rest-api-specs/blob/0c3632266c799502dccecd249377c606b3be9ab3/specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json#L680"}]},{"type":"Result","level":"Error","message":"The response is defined but without a default error response implementation.Consider adding it.'","code":"RequiredDefaultResponse","id":"R4010","docUrl":"https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/openapi-authoring-automated-guidelines.md#R4010"

**Comment by pilor** [comment]
>         "systemData": {

systemData is per resource.  It should not appear at the top level of a collection GET

---
Refers to: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:4837 in b6abd75. [](commit_id = b6abd75b43c569e9048f5add5920360003572159, deletion_comment = False)

**Comment by pilor** [comment]
>         "sshPassword": {

all properties that are secrets must be annoted as such.  Please also ensure they are not returned on non-POST responses and not returned in GETs.  [https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi013-secrets-must-be-annotated-with-x-ms-secret-and-x-ms-mutability](https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi013-secrets-must-be-annotated-with-x-ms-secret-and-x-ms-mutability)

---
Refers to: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:4887 in b6abd75. [](commit_id = b6abd75b43c569e9048f5add5920360003572159, deletion_comment = False)

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json
Comment (pilor) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:2951]: this description doesn't seem valid given CMK, etc... resides in this object

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json
Comment (pilor) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:3536]: Update this as well given that all storage accounts are encrypted now

Comment (pilor) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:3536]: make a pass through and ensure all the encryption related descriptions are accurate.  These description errors are littered throughout

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json
Comment (pilor) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:2661]: following up with the folks driving CMK via e-mail to ensure they are ok with this contract

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json
Comment (blueww) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:1764]: @jianyexi 
Why we have this check?
The check seems not fair , espicially for .net SDK. It will hide the real failure reason in deep level of excpetion, instead of show it in the exception message. 
Could we remove the check ?

Comment (jianyexi) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:1764]: This is required by ARM , may be dotnet SDK is special , for a consistent user experience for all the SDKs, we should respect default response

Comment (blueww) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:1764]: @jianyexi
Could you please give the reason why this is required by ARM?
currently from .net SDK point of view, the part is not necessary, and really take issue to .net SDK.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json
Comment (raomanus) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:3536]: Updated the description for the encryption properties in the swagger yo reflect current state.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json
Comment (raomanus) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:2951]: Will update this.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json
Comment (raomanus) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:2661]: Hey @pilor , we cleared this with Wei Wei over email previously. Is there any other concern here?

**Comment by zfchen95** [comment]
> > ```
> >     "systemData": {
> > ```
> 
> systemData is per resource. It should not appear at the top level of a collection GET
> 
> Refers to: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:4837 in [b6abd75](https://github.com/Azure/azure-rest-api-specs/commit/b6abd75b43c569e9048f5add5920360003572159). [](commit_id = [b6abd75](https://github.com/Azure/azure-rest-api-specs/commit/b6abd75b43c569e9048f5add5920360003572159), deletion_comment = False)

Removed system data from collection GET

**Comment by zfchen95** [comment]
> > ```
> >     "sshPassword": {
> > ```
> 
> all properties that are secrets must be annoted as such. Please also ensure they are not returned on non-POST responses and not returned in GETs. https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi013-secrets-must-be-annotated-with-x-ms-secret-and-x-ms-mutability
> 
> Refers to: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:4887 in [b6abd75](https://github.com/Azure/azure-rest-api-specs/commit/b6abd75b43c569e9048f5add5920360003572159). [](commit_id = [b6abd75](https://github.com/Azure/azure-rest-api-specs/commit/b6abd75b43c569e9048f5add5920360003572159), deletion_comment = False)

add the x-ms-secret tag

**Comment by zfchen95** [comment]
Can we have an exception on the modelValidation check? Because sshPassword is marked as secret, the model validation checks failed for these two examples: LocalUserCreate and LocalUserRegeneratePassword. It is returned in LocalUserCreate because the password is generated by the server and it is only returned at the creation time. For LocalUserRegeneratePassword, it is a POST action.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json
Comment (blueww) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:3684]: Why user `"modelAsString": true`?
If there are any possible new value to the enum in the future, use `"modelAsString": false` can avoid breaking.

Comment (zfchen95) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:3684]: @raomanus, could you please check?

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json
Comment (pilor) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:2661]: CMK folks have indicated this should be named "federatedClientId".  Getting closure from them on the e-mail thread

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json
Comment (zfchen95) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:3684]: @blueww, why "modelAsString": false can avoid SDK breaking? I am seeing other enums are modeled as strings.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json
Comment (raomanus) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:2661]: This is incorrect. Storage requires the property name to be "federatedIdentityClientId" and not "federatedClientId". Can you please let me know who told you this? I'm not aware of any email thread for this discussion.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json
Comment (blueww) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:3684]: My fault. Yes, should use `"modelAsString": true` to avoid SDK breaking.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json
Comment (blueww) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:4927]: Could user specify both blob and file, like "blob,file"?
As "resourceName" should be a container name or share name. so looks the permission scope can only include 1 resource, either in blob or in file.
If so, we might should change it to enum to help user input it more easily.

**Review by blueww — APPROVED** [review]
I am good with the change.
It's better if we can resolve the default error handling to make it work well in .net SDK.

**Comment by raosuhas** [comment]
>           "type": "string",

Readonly secrets should not be part of the payload since they are not returned on GET and cannot be set either. 

---
Refers to: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json:4888 in c67f05d. [](commit_id = c67f05df28b574ff0c038879d370f24078a31606, deletion_comment = False)

**Comment by zfchen95** [comment]
> Readonly secrets should not be part of the payload since they are not returned on GET and cannot be set either.

Removed the secret tag, since these properties are generated by the server and have to be returned in the PUT response. They are not returned in the GET response anyway.

**Comment by RamyasreeChakka** [comment]
@zfchen95 - Removing secret tag from swagger is not the proper way to fix the security concern. The secret should never be exposed in PUT/GET responses. You need to expose the secrets only in POST API. Please fix the issue. Without fixing the issue, the PR cannot be approved.

---
In reply to: [982901908](https://github.com/Azure/azure-rest-api-specs/pull/16659#issuecomment-982901908) [](http://example.com/codeflow?ancestors=982901908,982901908)

**Comment by zfchen95** [comment]
> @zfchen95 - Removing secret tag from swagger is not the proper way to fix the security concern. The secret should never be exposed in PUT/GET responses. You need to expose the secrets only in POST API. Please fix the issue. Without fixing the issue, the PR cannot be approved.
> 
> In reply to: [982901908](https://github.com/Azure/azure-rest-api-specs/pull/16659#issuecomment-982901908)

It is fixed. Please check my last commit. I have removed the secret from the PUT response

**Comment by zfchen95** [comment]
There is one check that failed in the modelValidation. But it seems to be wrong

Secret property "sshPassword", cannot be sent in the response.
Url: Microsoft.Storage/stable/2021-08-01/storage.json#L4885
JsonUrl: stable/2021-08-01/examples/LocalUserRegeneratePassword.json#L25

sshPassword can only be regenerated and returned in this POST action, similar to regenerate account key.

**Comment by RamyasreeChakka** [comment]
> removed

I see you removed the secrets in PUT response in the example file. Can you please confirm your service actual response doesn't contain the secrets?

**Comment by RamyasreeChakka** [comment]
> 

@zfchen95 I see you removed the secrets in PUT response in the example file. Can you please confirm your service actual response doesn't contain the secrets?

**Comment by zfchen95** [comment]
> > 
> 
> @zfchen95 I see you removed the secrets in PUT response in the example file. Can you please confirm your service actual response doesn't contain the secrets?

We will remove the secret from the PUT response on the server side.

**Comment by zfchen95** [comment]
work item for tracking [Bug 12739784: Do not return secret in Put/Pacth local users](https://msazure.visualstudio.com/DefaultCollection/One/_workitems/edit/12739784)

**Comment by RamyasreeChakka** [comment]
> 

@zfchen95 - Perfect! Thanks for making the change on server side.
As discussed offline, please remove the secret property from PUT definition response model in swagger spec. Do not use the same resource model for PUT and POST responses. Use a different model for POST and include secret details only in the post.

**Comment by zfchen95** [comment]
Another item to track server-side changes https://msazure.visualstudio.com/DefaultCollection/One/_workitems/edit/12740091

**Comment by RamyasreeChakka** [comment]
@zfchen95 - Thanks for making the suggested changes. I reviewed the remaining changes added after Suhas's review. LGTM. Signing off from ARM side.

**Comment by jianyexi** [comment]
@zfchen95  could you fix the https://github.com/Azure/azure-rest-api-specs/pull/16659/checks?check_run_id=4412720413 , other s look fine

**Comment by jianyexi** [comment]
@weidongxu-microsoft  @qiaozha  @msyyc could you help confirm the SDK generation failures

@zhenglaizhang  could you help confirm ApiDocPreview failure

**Comment by weidongxu-microsoft** [comment]
@jianyexi Java failed when generating SDK examples, so code is likely good. I will check this later. This should not block the PR.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/blob.json
Comment (ArcturusZhang) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/blob.json:1578]: Could you please apply the changes in this PR to your swagger to keep things consistent?
https://github.com/Azure/azure-rest-api-specs/pull/17027/files

File: specification/storage/resource-manager/readme.md
Comment (ArcturusZhang) [inline specification/storage/resource-manager/readme.md:30]: We should not adding a new default, instead we should modify the existing default tag ( you can find it on line 86 of this file.) Keeping them both will lead to generation errors, like azure-sdk-for-go-track2.

Please either remove these lines, and make the corresponding modification on line 86, or remove line 84-87

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]
Please make sure to apply the same changes from https://github.com/Azure/azure-rest-api-specs/pull/17027 to your new api-version. This is important to keep our SDKs stable.

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**
